tidy3d/components/mode/simulation.py

```diff
@@ -0,0 +1,337 @@
+        """Handle the case where grid_spec is auto and wavelength is not provided."""
+        # this is handled instead post-init to ensure freqs is defined
+        return val
+
+    @cached_property
+    def grid(self) -> Grid:
+        """Grid spatial locations and information as defined by `grid_spec`.
+
+        Returns
+        -------
+        :class:`.Grid`
+            :class:`.Grid` storing the spatial locations relevant to the simulation.
+        """
+
+        # Add a simulation Box as the first structure
+        structures = [Structure(geometry=self.geometry, medium=self.medium)]
+        structures += self.structures
+
+        # make source for autogrid if needed
+        freqs = self.freqs
+        grid_spec = self.grid_spec
+        sources = []
+        if grid_spec.auto_grid_used and grid_spec.wavelength is None:
+            if not np.all(np.isclose(freqs, freqs[0])):
+                raise SetupError(
+                    "Multiple 'sim.freqs' are supplied. Please supply "
+                    "a 'wavelength' value for 'grid_spec' to control automatic "
+                    "grid generation."
+                )
+            plane = self.geometry
+            sources.append(
+                ModeSource(
+                    center=plane.center,
+                    size=plane.size,
+                    source_time=GaussianPulse(freq0=freqs[0], fwidth=0.1 * freqs[0]),
+                    direction="+",
+                    mode_spec=self.mode_spec,
+                )
+            )
+
+        grid = self.grid_spec.make_grid(
+            structures=structures,
+            symmetry=self.symmetry,
+            periodic=self._periodic,
+            sources=sources,
+            num_pml_layers=self.num_pml_layers,
+        )
+
+        return grid
+
+    def _to_fdtd_sim(self) -> Simulation:
+        """Convert :class:`.ModeSimulation` to :class:`.Simulation`.
+        This should only be used to obtain the same material properties
+        for mode solving or related purposes; the sources and monitors of the
+        resulting simulation are not meaningful."""
+
+        # source to silence warnings
+        plane = self.geometry
+        freq0 = self.freqs[0]
+        source_time = GaussianPulse(freq0=freq0, fwidth=0.1 * freq0)
+        source = ModeSource(
+            center=plane.center,
+            size=plane.size,
+            source_time=source_time,
+            direction="+",
+            mode_spec=self.mode_spec,
+        )
+        return Simulation(
+            center=self.center,
+            size=self.size,
+            medium=self.medium,
+            structures=self.structures,
+            symmetry=self.symmetry,
+            grid_spec=self.grid_spec,
+            boundary_spec=self.boundary_spec,
+            version=self.version,
+            subpixel=self.subpixel,
+            run_time=RUN_TIME,
+            sources=[source],
+            monitors=[],
```

Comment:
could we add some way to switch whether this makes a monitor or a source?

Reply:
Same comment as above. This function isn't meant for the user, it's just for internal use for generating material properties